api/v1/services/UserService.js

```diff
@@ -118,3 +118,13 @@ module.exports.resetPassword = function (user, password) {
 			return updated.save();
 		});
 };
+
+module.exports.deleteUser = function (user) {
+	if(user.hasRoles(utils.roles.ALL, false)) {
+		var message = "The provided user has a role and can not be deleted";
+		var source = "UserRole";
+		throw new errors.InvalidParameterError(message, source);
```

Comment:
This should be `return _Promise.reject(new errors.InvalidParameterError(...));` if you want to make failure test work like the other promise functions.